src/server/game/World/World.cpp

```diff
@@ -3000,8 +3000,12 @@ void World::SendServerMessage(ServerMessageType type, const char *text, Player*
 {
     WorldPacket data(SMSG_SERVER_MESSAGE, 50);              // guess size
     data << uint32(type);
-    if (type <= SERVER_MSG_STRING)
+
+    if (sServerMessagesStore.AssertEntry(type)->IsFormattableMessage())
```

Comment:
that's  very unstable because it only takes someone to use the wrong type and all will blow up. Better use a LookupEntry call here and just cancel this whole helper execution if the entry is not found

Reply:
I'm not sure what the desired behavior should be though. In the scenario you propose though, someone has written code that tries to send a message that doesn't exist. That calling code is technically wrong. But should it crash/assert? Should it silently ignore? Should it log an error?

I can change it but I am just curious what the best approach would be.

In a perfect world we would auto-generate these types of enums from the DBC somehow and wouldn't be able to send messages that don't exist.

Reply:
I would just look up the entry at the beginning of the function and if the entry does not exist, just throw an error log and block further execution. Since the core is being used by all different kinds of people and there is no documentation, having it user friendly should be prioritized imo

Reply:
As requested I have removed the strict assertions and warn/error log instead. I think a better approach would be to make the server message type an enum class and check at startup that entries exist for all enumerated values. But I wouldn't know how to do that in C++, in C# that'd be easily done.

Anyway, it is more friendly now I guess.